primeorder/src/projective.rs

```diff
@@ -258,13 +258,24 @@ where
 
 impl<C> Group for ProjectivePoint<C>
 where
-    Self: Double,
+    Self: GroupEncoding,
     C: PrimeCurveParams,
 {
     type Scalar = Scalar<C>;
 
     fn try_from_rng<R: TryRngCore + ?Sized>(rng: &mut R) -> core::result::Result<Self, R::Error> {
-        Ok(Self::GENERATOR * <Scalar<C> as Field>::try_from_rng(rng)?)
+        let mut bytes = <Self as GroupEncoding>::Repr::default();
+
+        loop {
+            rng.try_fill_bytes(bytes.as_mut())?;
+
+            // Ensure SEC1 tag is 0x02 or 0x03 (compressed w\ even vs odd y-coordinate respectively)
+            bytes.as_mut()[0] = (bytes.as_mut()[0] & 1) | 2;
+
+            if let Some(point) = Self::from_bytes(&bytes).into() {
+                return Ok(point);
+            }
+        }
```

Comment:
You could avoid the bounds change here by using `AffinePoint<C>::decompress` (via the `DecompressPoint` trait) which is a bit more direct than `from_bytes` and avoids anything to do with SEC1.

That also works for `k256`.

Reply:
Nice.

However, for the sign I needed to generate a extra random byte. WDYT?
https://github.com/RustCrypto/elliptic-curves/blob/50164762766b167fe361f50fb9da41d13c0a61cb/primeorder/src/projective.rs#L267-L278

Reply:
On that note, bounds still needed to change.